### PR TITLE
DOCSP-38517: Go language landing page

### PR DESCRIPTION
**Pull request by norareidy**
JIRA - https://jira.mongodb.org/browse/DOCSP-38508
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/landing/DOCSP-38517-go-landing-page/languages/golang/

### CODE REVIEW

File: source/languages/golang.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+===============
+MongoDB with Go
+===============
+
+.. introduction::
+   
+   Build Go applications using MongoDB's :atlas:`Atlas cloud database </>`. Several
+   community-built integrations as well as Open Telemetry support developers
+   looking to build the next generation of Go-based solutions backed by MongoDB.
```
Comment (mongoKart) [inline source/languages/golang.txt:13]: Hard to parse; "support" initially comes across as a noun, not a verb. also, no space in OpenTelemetry

```suggestion
   Build Go applications using MongoDB's :atlas:`Atlas cloud database </>`. OpenTelemetry
   and community-built integrations support developers
   looking to build the next generation of Go-based solutions backed by MongoDB.
```

File: source/languages/golang.txt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+Provides a high-level API for working with MongoDB databases consistent with
+other MongoDB language drivers.
+
+`Go Driver Documentation → <https://www.mongodb.com/docs/drivers/go/current/>`__
+
+Integrations
+------------
+
+`Open Telemetry → <https://opentelemetry.io/>`__
```
Comment (mongoKart) [inline source/languages/golang.txt:32]:
```suggestion
`OpenTelemetry → <https://opentelemetry.io/>`__
```

**Comment by caitlindavey** [comment]
LGTM
